### DISCUSSION TITLE
Parsed Material for tensor properties

### DISCUSSION THREAD

**Discussion by lekah** [idaholab/moose#20273] [Q&A General] [answered]
Hi,
my application requires something like ParsedMaterial but to initialize a tensor from arguments and other material properties (each component individually). I could not find such functionality in the code, but since this is maybe a common problem someone has already coded this up? Or I overlooked something?
Cheers

**Comment by GiudGiud** [comment, marked as answer]
@dschwen have you heard of a ParsedTensorMaterial out there?

Guillaume

**Reply by lekah** [reply]
That sounds like exactly what I need :)

**Reply by GiudGiud** [reply]
I dont think we have it. 
@lekah you still needing this?

Other solution is to simply code a regular tensor material, there are a few examples in the codebase.

**Reply by lekah** [reply]
Hi @GiudGiud , 
A regular tensor material would not have the dependencies on arguments of each component, right?
I've found a workaround for my usecase by adding different tensors together, with variable-dependent weights, aka. CompositeTensor.
But a ParsedTensorMaterial would be a nice functionality and still useful to me.

**Reply by GiudGiud** [reply]
hello

Well you can definitely create a new object in the source code and have any dependencies you want. I guess it's not clear what I meant by "regular"
I ll see if i can create one in the near future, I ll let you know here if I have time 

Guillaume

**Reply by lekah** [reply]
Thanks Guillaume,
I was thinking of writing it up, but I didn't find the time as well. It would also take me some time to understand and design how to interface with ParsedFunctions etc... (fairly new to MOOSE). And the workaround with the CompositeTensor is sufficient for my usecase, but not elegant. For newbie-users however, I think it would be a nice functionality.
